lib/plugin/compilers/i18n.json.coffee

```diff
@@ -4,124 +4,147 @@ helpers = share.helpers
 compilers = share.compilers
 compiler_configuration = share.compiler_configuration
 
-compilers.i18n_json = (compileStep) ->
-  compiler_configuration.registerInputFile(compileStep)
```

Comment:
not 100% sure if I understand what's going on, but if you want to share the state, you would need to make it a single class/dispatch compilation.